fastdialer/dialer_private.go

```diff
@@ -399,3 +411,21 @@ func (d *Dialer) handleDialError(err error, opts *dialOptions) error {
 	}
 	return errx
 }
+
+// closeAfterTimeout closes the sockets after the given duration unless the returned cancel function is called
+func closeAfterTimeout(d time.Duration, c ...io.Closer) context.CancelFunc {
+	handshakeDoneCtx, handshakeDoneCancel := context.WithCancel(context.Background())
+	t := time.NewTimer(d)
+	go func() {
+		select {
+		case <-t.C:
+			for _, cl := range c {
+				cl.Close()
+			}
```

Comment:
I think we should stop the timer after closing all the sockets, so that the go-routine could return and release resources. What do you think?

Reply:
Good point! I've pushed an update to address this